### PR TITLE
Player Inventory Base

### PR DESCRIPTION
**Pull request by KyriakosNU**

### CODE REVIEW

File: CSC8503/InventoryBuffSystem/PlayerInventory.h [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+		PlayerInventory::item GetRandomItemFromPool(unsigned int seed);
+	private:
+
+		std::vector<item> itemsInRandomPool;
+
+		std::map<item, float> itemInitDurationMap =
+		{
+			{disguise,10},{item2,4}
+		};
```
Comment (Nijelous) [inline CSC8503/InventoryBuffSystem/PlayerInventory.h:44]: mItemsInRandomPool
mItemInitDurationMap

File: CSC8503/InventoryBuffSystem/PlayerInventory.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
+	#include "PlayerInventory.h"
+
+	using namespace InventoryBuffSystem;
+
+	void PlayerInventory::Init()
+	{
+		for (int playerNo = 0; playerNo < 4; playerNo++)
```
Comment (Nijelous) [inline CSC8503/InventoryBuffSystem/PlayerInventory.cpp:7]: Probably should remove the "magic number" of 4, make it a constexpr of PLAYER_MAX

File: CSC8503/InventoryBuffSystem/PlayerInventory.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		{
+			for (int invSlot = 0; invSlot < 2; invSlot++)
```
Comment (Nijelous) [inline CSC8503/InventoryBuffSystem/PlayerInventory.cpp:9]: Same for this, you've even used it further down as MAX_INVENTORY_SLOTS you just need to add it here

File: CSC8503/InventoryBuffSystem/PlayerInventory.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+			{
+				mPlayerInventory[playerNo][invSlot] = none;
+			}
+		}
+
+	}
+
+	void PlayerInventory::AddItemToPlayer(item inItem, int playerNo)
+	{
+		for (int invSlot = 0; invSlot < 2; invSlot++)
```
Comment (Nijelous) [inline CSC8503/InventoryBuffSystem/PlayerInventory.cpp:19]: And here

**Review by Nijelous — CHANGES_REQUESTED** [review]
Again, overall code looks good, please make recommended changes

File: CSC8503/InventoryBuffSystem/PlayerInventory.h [hunk 5]
```diff
@@ -10,9 +10,10 @@ using namespace NCL;
 using namespace CSC8503;
 using namespace InventoryBuffSystem;
 
-//const int MAX_PLAYERS = 4;
+
 namespace InventoryBuffSystem
 {
+	const int MAX_PLAYERS = 4;
 	const int MAX_INVENTORY_SLOTS = 4;
```
Comment (Nijelous) [inline CSC8503/InventoryBuffSystem/PlayerInventory.h:17]: Needs to be 2 slots, not 4

**Review by Nijelous — CHANGES_REQUESTED** [review]

**Review by Nijelous — APPROVED** [review]

File: CSC8503/InventoryBuffSystem/PlayerInventory.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
+	#include "PlayerInventory.h"
+
+	using namespace InventoryBuffSystem;
+
+	void PlayerInventory::Init()
+	{
```
Comment (fm19121) [inline CSC8503/InventoryBuffSystem/PlayerInventory.cpp:6]: Looks good, but in our code stander. The curly bracket should be on the same line with the function, like:
void PlayerInventory::Init(){
         .......
         ........
}

**Review by fm19121 — APPROVED** [review]

**Pull request merged**